### PR TITLE
Add question box v4 to support relative container

### PR DESCRIPTION
**Pull request by ArthurJordao**
# :wrench: Add question box v4 to support relative container

## Context

The parent relative div does not support the question box, this adds support to add the parent div to calculate the offsite by the parent relative container.

## :framed_picture: What does this change look like?
image

- Follow the the [Guidelines for an optimal design ping](https://paper.dropbox.com/doc/Guidelines-for-Sharing-User-Facing-Changes-with-Design--BpL8hpJLMugy6033aT5m0JdaAg-bdKGQtYH9qO9I00hUkA6k) to add helpful screenshots/videos for design.

### CODE REVIEW

File: src/Nri/Ui/QuestionBox/V4.elm [hunk 1]
```diff
@@ -0,0 +1,439 @@
...
+            -- position in the middle of the block
+            (block.element.x + (block.element.width / 2))
+                - -- against the middle of the question box
+                  (questionBox.element.width / 2)
+                - (case container of
+                    Just containerElement ->
+                        containerElement.element.x
+
+                    Nothing ->
+                        0
```
Comment (tesk9) [inline src/Nri/Ui/QuestionBox/V4.elm:239]: ✏️  Maybe a comment here explaining why this is necessary?

**Review by tesk9 — COMMENTED** [review]
I have some suggestions on https://github.com/NoRedInk/noredink-ui/pull/1268 to hopefully make it more clear what happens when the question box is positioned at the right side of the content.

Which version of [Stacey's suggestions](https://noredink.slack.com/archives/C0398SY837X/p1675979589134689?thread_ts=1675978528.109699&cid=C0398SY837X) is this going for? Currently, it seems to me that the content is constrained to the non-static ancestor on the left side, but it does still overlap on the right side.

image

**Comment by ArthurJordao** [comment]
@tesk9 oh I forgot the case for right size on the viewport, I just fixed it!

File: src/Nri/Ui/QuestionBox/V4.elm [hunk 2]
```diff
@@ -228,18 +228,35 @@ viewPointingTo config blockId measurements =
...
+                    case container of
+                        Just containerElement ->
+                            containerElement.element.x
+
+                        Nothing ->
+                            0
+
+                -- calculate the max position of the question box, so it will not render outside the viewport
+                maybeMaxPosition =
+                    container |> Maybe.map (\containerElement -> containerElement.element.x + containerElement.element.width - questionBox.element.width - offset) |> Debug.log "maxPosition"
```
[the remark this answers is not in the document]
Comment (ArthurJordao) [reply src/Nri/Ui/QuestionBox/V4.elm]: Hey, @tesk9  I just merged your PR, I was not aware of the xOffsetPx, this looks so much better! Thank you a lot for your help! Could you take a last look at the PR to see if everything is ok? I will try to merge it after and make a new release to finish the work on the migration!

**Review by tesk9 — APPROVED** [review]
🎉 🐼 🎉

**Pull request merged**